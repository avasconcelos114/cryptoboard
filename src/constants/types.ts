export interface Theme {
    base_font_color: string,
    drop_shadow: string,
    drop_shadow_hover: string,
    header_background: string,
    loading_background: string,
    main_background: string,
    news_background: string,
    news_category_line: string,
    news_category_backgrounds: string[],
    sidebar_background: string,
    sidebar_font_color: string,
}

export interface Menu {
    title: string,
    route: string,
}

export interface News {
    id: string,
    title: string,
    imageurl: string,
    url: string,
    categories: string,
    upvotes: string,
    downvotes: string,
}