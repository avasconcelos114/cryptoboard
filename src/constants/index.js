export default {
    menuItems: [
        {title: 'Dashboard', url: '/dashboard'},
        {title: 'Top List', url: '/toplists'},
        {title: 'All Coins', url: '/coinlist'},
        {title: 'News', url: '/news'},
    ],
}