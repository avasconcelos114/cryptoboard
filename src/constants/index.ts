/* 
   Under normal circumstances I'd be serving a backend server in the same endpoint
   and authenticating the client to retrieve a valid session token. But given the fact that I'm using
   a public API with 100K monthly free calls, I'm just leaving it in the open to allow the app to run
   off anywhere equally.
*/
export const api = {
    token: '328615ca08e935e05a0ddd27d5c746093b35cb0387ecb5b64ff29bc7e63094cb',
    endpoint: 'https://min-api.cryptocompare.com',
};

export const themes = {
    dark: {
        baseFontColor: 'rgba(255, 255, 255, 0.87)',
        mediumFontColor: 'rgba(255, 255, 255, 0.60)',
        disabledFontColor: 'rgba(255, 255, 255, 0.38)',
        dropShadow: 'rgba(214,214,214,0.8)',
        dropShadowHover: 'rgba(214,214,214,1)',
        loadingBackground: '#121212',
        mainBackground: '#121212',
        cardBackground: 'rgba(255, 255, 255, 0.07)',
        cardLineColor: 'rgba(255, 255, 255, 0.11)',
        newsCategoryBackgrounds: [
            '#03A9F4',
            '#26C6DA',
            '#26A69A',
            '#3F51B5',
            '#7E57C2',
            '#9C27B0',
            '#4527A0',
        ],
        sidebarBackground: 'rgba(255, 255, 255, 0.05)',
    },
    light: {
        baseFontColor: 'rgba(0, 0, 0, 0.87)',
        mediumFontColor: 'rgba(0, 0, 0, 0.60)',
        disabledFontColor: 'rgba(0, 0, 0, 0.38)',
        dropShadow: 'rgba(214,214,214,0.4)',
        dropShadowHover: 'rgba(214,214,214,0.6)',
        loadingBackground: '#FAFAFA',
        mainBackground: '#FAFAFA',
        cardBackground: 'rgba(255, 255, 255, 1)',
        cardLineColor: '#E0E0E0',
        newsCategoryBackgrounds: [
            '#03A9F4',
            '#26C6DA',
            '#26A69A',
            '#3F51B5',
            '#7E57C2',
            '#9C27B0',
            '#4527A0',
        ],
        sidebarBackground: '#FFFFFF',
    },
};

export const menuItems = [
    { title: 'Dashboard', route: 'dashboard' },
    { title: 'Top List', route: 'top_coins' },
    // {title: 'All Coins', route: 'all_coins'},
    { title: 'News', route: 'news' },
];

export default {
    themes,
    menuItems,
};
