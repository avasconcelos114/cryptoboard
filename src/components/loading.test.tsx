import React from 'react';
import Loading from './loading';
import constants from '../constants';

import { shallow } from 'enzyme';

const props = {
    theme: constants.themes.light,
    isLoading: false,
    id: 'test_loading_component'
}

const setup = (props: any) => {
    const component = shallow(
        <Loading {...props} />
    )
    return {
        component: component
    }
}

describe('Loading', () => {
    it('renders without crashing', () => {
        const { component } = setup(props)
        expect(component).toMatchSnapshot()
    });
    
    it('should return a null component when isLoading is false', () => {
        const {component} = setup(props)
        expect(component.html()).toBeNull()
    });
    
    it('should render loading component when isLoading is true', () => {
        const newProps = {...props, isLoading: true}
        const { component } = setup(newProps)
        expect(component.html()).not.toBeNull()
    });
})
