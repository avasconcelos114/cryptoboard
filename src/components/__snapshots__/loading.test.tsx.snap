// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders without crashing 1`] = `ShallowWrapper {}`;
